mini-projects/00-jq-rs/src/processor/printer.rs

```diff
@@ -0,0 +1,424 @@
+use serde_json::Value;
+//All of the types of outputs our filtering/operations could leave for us to print
+pub enum PrinterInputs {
+    String(String),
+    Value(Value),
+    Number(usize),
+}
+//may be neater to compact inputs into a tuple
+pub fn print(
+    input: PrinterInputs,
+    color: bool,
+    sort: bool,
+    indent: u32,
+    compact: bool,
+    jq_color: &str,
+) {
+    //convert the given JQ_Env string into a vector of numbers representing the color/format
+    let jq_vec = parse_jq_env(jq_color);
+    //while match statements may not be super scalable, short of adding a new type of value JSON supports
+    //we shouldn't need to add more conditions here
+    match input {
+        PrinterInputs::String(string) => {
+            println!("{}", print_string(string, color, jq_vec[4].0, jq_vec[4].1))
+        }
+        PrinterInputs::Number(num) => {
+            println!("{}", print_numb(num as f64, color, jq_vec))
+        }
+        PrinterInputs::Value(val) => {
+            println!("{}", print_value(val, color, sort, indent, compact, jq_vec));
+        }
+    };
+}
+fn print_value(
+    input: Value,
+    color: bool,
+    sort: bool,
+    indent: u32,
+    compact: bool,
+    jq_vec: Vec<(&str, &str)>,
+) -> String {
```

Comment:
I think having a print trait that multiple different print structs imply could have cleaned up some of this logic - without it, seems like there's a lot of branching within each subsection of this function